cmd/core-data/main.go

```diff
@@ -45,6 +47,20 @@ func main() {
 	flag.Parse()
 
 	httpServer := handlers.NewServerBootstrap(data.LoadRestRoutes())
+
+	bootstrapHandlers := []interfaces.BootstrapHandler{
+		data.NewServiceInit(&httpServer).BootstrapHandler,
+		telemetry.BootstrapHandler,
+		httpServer.Handler,
+		handlers.NewStartMessage(clients.CoreDataServiceKey, edgex.Version).Handler,
+	}
+
+	// Unless the secret store functionality has been explicitly disabled, add the SecretClient bootstrap handler as the
+	// first handler since it will be needed by other bootstrap handlers.
+	if env := os.Getenv(internal.SecretStore); env != "false" {
+		bootstrapHandlers = append([]interfaces.BootstrapHandler{handlers.SecretClientBootstrapHandler}, bootstrapHandlers...)
```

Comment:
Prefer `bootstrapHandlers = append(bootstrapHandlers, handlers.SecretClientBootstrapHandler)`

Reply:
This will go away based on you other comment suggesting that the `handlers.SecretClientBootstrapHandler` should always be included in the bootstrap handlers provided to the `Run` function, and within the handler we should decided to determine if we should create the client. However, the reason why I am doing:
```
bootstrapHandlers = append([]interfaces.BootstrapHandler{handlers.SecretClientBootstrapHandler}, bootstrapHandlers...)
```

rather than:
```
bootstrapHandlers = append(bootstrapHandlers, handlers.SecretClientBootstrapHandler)
```

is because we want the `handlers.SecretClientBootstrapHandler` to be the first handler in the slice that we are passing to the `bootstrap.Run` so that it executes first. The other implementation would add the `handlers.SecretClientBootstrapHandler` to the end of the slice, thus getting executed last. This is a non-issue anyway as the logic for structuring the slice will go away.

Reply:
Doh!  Thank you!